src/main/java/com/tabnine/binary/requests/notifications/HoverBinaryRequest.kt

```diff
@@ -0,0 +1,19 @@
+package com.tabnine.binary.requests.notifications
+
+import com.tabnine.binary.BinaryRequest
+import com.tabnine.binary.exceptions.TabNineInvalidResponseException
+
+class HoverBinaryRequest : BinaryRequest<HoverBinaryResponse> {
+    override fun response(): Class<HoverBinaryResponse> {
+        return HoverBinaryResponse::class.java
+    }
+
+    override fun serialize(): Any {
+        return mapOf("Hover" to Any())
+    }
+
+    override fun shouldBeAllowed(e: TabNineInvalidResponseException): Boolean {
+        //allow null responses.
+        return e.rawResponse.isPresent && "null".equals(e.rawResponse.get())
```

Comment:
I think this could be written a bit differently to make it easier to read:
```kotlin
        return e.rawResponse.map(it == "null").orElse(false)
```

Reply:
Changed.